src/base/matrix.rs

```diff
@@ -161,16 +157,18 @@ pub type MatrixCross<T, R1, C1, R2, C2> =
 #[cfg_attr(
     feature = "rkyv-serialize-no-std",
     derive(Archive, rkyv::Serialize, rkyv::Deserialize),
-    archive(
-        as = "Matrix<T::Archived, R, C, S::Archived>",
-        bound(archive = "
-        T: Archive,
-        S: Archive,
-        With<PhantomData<(T, R, C)>, CustomPhantom<(Archived<T>, R, C)>>: Archive<Archived = PhantomData<(Archived<T>, R, C)>>
-    ")
+    rkyv(
+        compare(PartialEq, PartialOrd),
+        derive(Debug),
+        archive_bounds(
+            <S as Archive>::Archived: core::fmt::Debug,
```

Comment:
I feel like this is a bit constraining ? This is only needed for automated derive (needed for tests)
 
See https://github.com/dimforge/nalgebra/pull/1502/files#r2025182401